[3.8] bpo-32912: Revert warnings for invalid escape sequences.

serhiy-storchaka opened the pull request:
DeprecationWarning will continue to be emitted for invalid
escape sequences in string and bytes literals in 3.8.
SyntaxWarning will be emitted in 3.9.


<!-- issue-number: [bpo-32912](https://bugs.python.org/issue32912) -->
https://bugs.python.org/issue32912
<!-- /issue-number -->

serhiy-storchaka committed:
bpo-32912: Revert warnings for invalid escape sequences.

DeprecationWarning will continue to be emitted for invalid
escape sequences in string and bytes literals in 3.8.
SyntaxWarning will be emitted in 3.9.

jdemeyer commented on Doc/reference/lexical_analysis.rst:
```diff
@@ -594,11 +594,8 @@ escape sequences only recognized in string literals fall into the category of
-      Unrecognized escape sequences produce a :exc:`SyntaxWarning`.  In
-      some future version of Python they will be a :exc:`SyntaxError`.
+      Unrecognized escape sequences produce a DeprecationWarning.  In
+      Python 3.9 they will be a SyntaxError.
```
I don't think we should promise to raise `SyntaxError` in 3.9. This issue will no doubt lead to discussions again in the 3.9 cycle, so I wouldn't mention 3.9 and just drop that last sentence.

serhiy-storchaka replied:
This PR has been created in the assumption that SyntaxError will be kept in 3.9. If we decide other, it should be rebased against the master branch.

jdemeyer replied:
Are you confusing `SyntaxWarning` with `SyntaxError`?

serhiy-storchaka replied:
Good catch!

gpshead replied:
I modified the wording to not say "3.9" but instead say "future Python version" as we clearly don't have consensus on when and how this will move beyond DeprecationWarning so it is best not to promise a timeline.

serhiy-storchaka committed:
Fix a typo.

gpshead committed:
update the wording to not promise 3.9

and mention that SyntaxError will happen after SyntaxWarning.  we're being non-specific on purpose here as we haven't worked out the plan to make this happen without causing end user pain yet.  right now most code owners do not see the DeprecationWarning but when turned into a SyntaxWarning, _users_ of libraries see it more often than owners.  So it is a negative experience all around.

CuriousLearner reviewed (approved):
I tend to agree with @gpshead on applying this to master and then backporting to 3.8
